internal/security/credentials.go

```diff
@@ -47,30 +52,81 @@ func (s *SecretProvider) GetDatabaseCredentials(database db.DatabaseInfo) (commo
 // specified path will be returned.
 func (s *SecretProvider) GetSecrets(path string, keys ...string) (map[string]string, error) {
 
+	if cachedSecrets := s.getCachedSecrets(path, keys...); cachedSecrets != nil {
+		return cachedSecrets, nil
+	}
+
+	if !s.isSecurityEnabled() {
+		s.loggingClient.Debug(fmt.Sprintf("Writable.InsecureSecrets: %v", s.configuration.Writable.InsecureSecrets))
+		return s.getInsecureSecrets(path, keys...)
+	}
+
+	newSecrets, err := s.secretClient.GetSecrets(path, keys...)
+	if err != nil {
+		return nil, err
+	}
+
+	s.secrets[path] = newSecrets
+	return s.secrets[path], nil
+}
+
+// GetInsecureSecrets retrieves secrets from the Writable section of the configuration
+// path specifies the type or location of the secrets to retrieve.
+// keys specifies the secrets which to retrieve. If no keys are provided then all the keys associated with the
+// specified path will be returned.
+func (s *SecretProvider) getInsecureSecrets(path string, keys ...string) (map[string]string, error) {
+	secrets := make(map[string]string)
+	pathExists := false
+	var missingKeys []string
+
+	for _, insecureSecrets := range s.configuration.Writable.InsecureSecrets {
+		if insecureSecrets.Path == path {
+			pathExists = true
+			for _, key := range keys {
+				value, keyExists := insecureSecrets.Secrets[key]
+				if !keyExists {
+					missingKeys = append(missingKeys, key)
+					continue
+				}
+				secrets[key] = value
+			}
+		}
+	}
+
+	if len(missingKeys) > 0 {
+		err := fmt.Errorf("No value for the keys: [%s] exists", strings.Join(missingKeys, ","))
+		return nil, err
+	}
+
+	if !pathExists {
+		// if path is not in secret store
+		err := errors.New("Error found on handling secrets from underlying data-store: Received a '404' response from the secret store")
```

Comment:
Please re-word the error message- 404 status code from a configuration file is kind of misleading.